distributed/diagnostics/nvml.py

```diff
@@ -281,18 +316,22 @@ def get_device_mig_mode(device):
         A ``list`` with two integers ``[current_mode, pending_mode]``.
     """
     init_once()
-    try:
-        device_index = int(device)
-        handle = pynvml.nvmlDeviceGetHandleByIndex(device_index)
-    except ValueError:
-        uuid = device if isinstance(device, bytes) else bytes(device, "utf-8")
-        handle = pynvml.nvmlDeviceGetHandleByUUID(uuid)
+    handle = _get_handle(device)
     try:
         return pynvml.nvmlDeviceGetMigMode(handle)
     except pynvml.NVMLError_NotSupported:
         return [0, 0]
 
 
+def _get_handle(device):
+    try:
+        device_index = int(device)
+        return pynvml.nvmlDeviceGetHandleByIndex(device_index)
+    except ValueError:
+        uuid = device if isinstance(device, bytes) else bytes(device, "utf-8")
+        return pynvml.nvmlDeviceGetHandleByUUID(uuid)
+
+
 def _get_utilization(h):
```

Comment:
It looks like this method is returning `None` when using MIG devices which eventually results in the `int()` cast failing later on.

Reply:
It looks like this is to be expected. The documentation mentions that utilization rates is not supported on MIG https://docs.nvidia.com/deploy/nvml-api/group__nvmlDeviceQueries.html#group__nvmlDeviceQueries_1g540824faa6cef45500e0d1dc2f50b321